doc/OSAL-Configuration-Guide.md

```diff
@@ -287,74 +216,106 @@ available via the respective Linux distribution package management
+| `ENABLE_UNIT_TESTS`  | TRUE or FALSE  |  Optional; defaults to "FALSE" if not specified. If set to TRUE, the included unit test code will also be built in addition to the runtime library. |
+| `OSAL_USER_C_FLAGS`  | Any valid switches for the compiler in use.  | Optional; the user may specify any arbitrary compiler switches to use.  |
 
 
 It is important to note that the values specified for these variables
 are **automatically cached** by the cmake build system. It is only
 necessary to specify these values when first provisioning/configuring a
-build; these are *not* required when simply building the binaries. 
+build; these are *not* required when simply building the binaries.
 
 This caching function removes the need for environment variables to be
-set as in the "setvars.sh" file in the classic build. The cmake build
-does not require the user to set environment variables; all necessary
-context information is automatically stored in the cache when the build
-is first provisioned. 
+set in the development shell. The generated makefiles do not require the
+user to set any environment variables, as all necessary context information
+is automatically stored in the cache when the build is first provisioned.
 
 ### Setting up a standalone OSAL build
 
 The OSAL may be built standalone in order to evaluate the library for a
-particular target and/or execute the included unit tests. 
+particular target and/or execute the included unit tests.
 
 In the cmake build system, all generated files are placed in a dedicated
 "binary" directory that is separate from the source tree. To provision a
 build, the user must first create the binary directory by issuing the
 "mkdir" command (on Linux), preferably outside the OSAL source tree.
 Then, the "cmake" provisioning tool is invoked to generate the actual
-makefiles, supplying values for the required variables: 
+makefiles, supplying values for the required variables.
+
+The following example will generate a build tree that allows testing
+of the OSAL library on a standard Linux workstation:
 
 ```
 $ mkdir build
 $ cd build
-$ cmake –DOSAL_SYSTEM_OSTYPE=posix –DOSAL_SYSTEM_BSPTYPE=pc-linux \
-  –DENABLE_UNIT_TESTS=TRUE –DOSAL_INCLUDEDIR=/path/to/user/config \
-  /path/to/osal/source
-
+$ cmake \
+    -DCMAKE_BUILD_TYPE=debug \
+    -DENABLE_UNIT_TESTS=true \
+    -DOSAL_SYSTEM_BSPTYPE=generic-linux \
+    -DOSAL_CONFIG_DEBUG_PERMISSIVE_MODE=true \
+     /path/to/osal/source
 ```
 
-
 The cmake provisioning tool generates standard makefiles in the build
 directory. To build the OSAL binaries, simply run "make" in the build
-directory. 
+directory.
+
+Note that the OSAL is not typically built as a standalone library; although a standalone
+build can be done for testing and development purposes, it is expected that the OSAL will
+be utilized as a component in a larger application.
 
 ### Integrating OSAL into a larger build
```

Comment:
Love this, ping @ejtimmon for reference to reviving the tasking examples and adding c++ hello world